### ISSUE TITLE
ModuleParameterRange symmetric curve=1 maps only the upper half and breaks round trips

### ISSUE THREAD

**Issue by turian** [torchsynth/torchsynth#499]
## Reproduction

On current `main` (`2b0964d4c6c3d472a2a0d54d91b408caaeffca6d`), the documented linear symmetric range does not map normalized endpoints/center to the physical endpoints/center:

```python
import torch
from torchsynth.parameter import ModuleParameterRange

r = ModuleParameterRange(-10, 10, curve=1, symmetric=True)
n = torch.tensor([0.0, 0.5, 1.0])
physical = r.from_0to1(n)
print(physical.tolist())
print(r.to_0to1(physical).tolist())
```

Observed:
```text
[0.0, 5.0, 10.0]
[0.5, 0.75, 1.0]
```

Expected:
```text
[-10.0, 0.0, 10.0]
[0.0, 0.5, 1.0]
```

Independently reproduced while auditing TorchSynth as a reference for a hardware port, using CPU float32, Python 3.13.2 / PyTorch 2.14.0. The imported module path and source pin were verified. This example is exact simple arithmetic, not the separate CPU/GPU or batch-shape floating-point drift discussed in #256.

## Cause

In [parameter.py:78–87](https://github.com/torchsynth/torchsynth/blob/2b0964d4c6c3d472a2a0d54d91b408caaeffca6d/torchsynth/parameter.py#L78), `dist = 2 * normalized - 1` is computed, but `normalized` is only replaced by the centered result when `curve != 1.0`. For `curve == 1`, the final symmetric mapping therefore uses the original [0,1] coordinate instead of the centered [-1,1] coordinate. For any increasing [min,max] range, this maps into only its upper half.

The class documentation says curve one is linear, and `to_0to1` uses the centered convention, so endpoint and round-trip behavior disagree.

## Suggested regression coverage

- Symmetric curve=1 and 1.0: endpoints, center, and interior values.
- Signed and nonsigned physical ranges (for example [-10,10] and [2,6]).
- Forward/inverse round trips.
- Existing symmetric nonlinear and nonsymmetric behavior remains unchanged.

A narrow fix can use the centered coordinate for the linear symmetric branch; no broader parameter API change is needed.

## Impact / compatibility note

No range in the pinned **default Voice** uses `symmetric=True, curve=1`, so this report is **not** a claim that default Voice audio is wrong or that the hardware-port reference needs to be changed. The downstream port deliberately retains its pinned upstream behavior. Custom modules/ranges using this public configuration are affected; a fix would change their generated parameters and should be documented accordingly.

Searched issues/PRs for symmetric, from_0to1 and linear/curve variants before filing; related parameter-design issues and #256 do not cover this linear symmetric branch.
